pkg/cmd/roachtest/operations/backup_restore.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package operations
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operation"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestflags"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+)
+
+type backupRestoreCleanup struct {
+	db string
+}
+
+func (cl *backupRestoreCleanup) Cleanup(
+	ctx context.Context, o operation.Operation, c cluster.Cluster,
+) {
+	conn := c.Conn(ctx, o.L(), 1, option.VirtualClusterName(roachtestflags.VirtualCluster))
+	defer conn.Close()
+
+	o.Status(fmt.Sprintf("dropping newly created db %s", cl.db))
+	_, err := conn.ExecContext(ctx, fmt.Sprintf("DROP DATABASE %s CASCADE", cl.db))
+	if err != nil {
+		o.Fatal(err)
+	}
+}
+
+func runBackupRestore(
+	ctx context.Context, o operation.Operation, c cluster.Cluster,
+) registry.OperationCleanup {
+	// This operation looks for the district table in a database named cct_tpcc or tpcc.
+	rng, _ := randutil.NewPseudoRand()
+	dbWhitelist := []string{"cct_tpcc", "tpcc"}
+	conn := c.Conn(ctx, o.L(), 1, option.VirtualClusterName(roachtestflags.VirtualCluster))
+	defer conn.Close()
+	dbs, err := conn.QueryContext(ctx, "SELECT database_name FROM [SHOW DATABASES]")
+	if err != nil {
+		o.Fatal(err)
+	}
+	var dbName string
+	for dbs.Next() {
+		var dbStr string
+		if err := dbs.Scan(&dbStr); err != nil {
+			o.Fatal(err)
+		}
+		for i := range dbWhitelist {
+			if dbWhitelist[i] == dbStr {
+				// We found a db in the whitelist.
+				dbName = dbStr
+				break
```

Comment:
Nit / hint: You can use labels to break out of the outer for loop (and avoid the additional `dbName != ""` check in the outer loop): https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/urlcheck/lib/urlcheck/urlcheck.go#L188
https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/urlcheck/lib/urlcheck/urlcheck.go#L206
doc ref: https://go.dev/ref/spec#Break_statements (break statements)